gRPC/Client Examples/ImpedanceCheckApp/MainWindow.xaml.cs

```diff
@@ -0,0 +1,110 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Timers;
+using System.Windows;
+using System.Windows.Controls;
+using System.Windows.Data;
+using System.Windows.Documents;
+using System.Windows.Forms.DataVisualization;
+using System.Windows.Input;
+using System.Windows.Media;
+using System.Windows.Media.Imaging;
+using System.Windows.Navigation;
+using System.Windows.Shapes;
+
+namespace ImpedanceCheckApp
+{
+    /// <summary>
+    /// Interaction logic for MainWindow.xaml
+    /// </summary>
+
+    public partial class MainWindow : Window
+    {
+        private RealtimeGraphing.BICManager aBICManager;
+        private bool connectState = false;
+
+        // Logging
+        FileStream impedFileStream;
+        StreamWriter impedFileWriter;
+
+        public MainWindow()
+        {
+            InitializeComponent();
+        }
+
+        private void MainWindow_Loaded(object sender, RoutedEventArgs e)
+        {
+
+            // Establish connection
+            aBICManager = new RealtimeGraphing.BICManager((int)MainWindow1.Width, true); // additional parameter that is set to default otherwise?
+            connectState = aBICManager.BICConnect();
+
+            // Perform impedance check upon startup
+            performImpCheck(connectState);
+        }
+
+        private void MainWindow_Closed(object sender, EventArgs e)
+        {
+            aBICManager.Dispose();
+        }
+
+        private void btn_repeatcheck_Click(object sender, RoutedEventArgs e)
+        {
+            ImpedanceOutputConsole.Inlines.Add("Repeating impedance check... \n");
```

Comment:
Why are we disposing the BIC and re-initializing it to do impedance check? Can impedance checks ONLY be done at start-up?